### PR TITLE
[4/n] [RFC] add launch multiple runs backend functionality

### PR DESCRIPTION
**Pull request by dliu27**
Linear: https://linear.app/dagster-labs/issue/FE-659/add-launch-all-backend-functionality

## Summary & Motivation
Add a backend GraphQL mutation to handle launching multiple runs

## How I Tested These Changes
`pytest python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_run_launcher.py`
50 passed, 18 warnings in 206.16s (0:03:26)

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/graphql/schema.graphql [hunk 1]
```diff
@@ -3002,6 +3010,10 @@ type LaunchRunSuccess implements LaunchPipelineRunSuccess {
   run: Run!
 }
 
+type LaunchMultipleRunsSuccess {
```
Comment (dliu27) [inline js_modules/dagster-ui/packages/ui-core/src/graphql/schema.graphql:3013]: rm

File: python_modules/dagster-graphql/dagster_graphql/schema/roots/mutation.py [hunk 2]
```diff
@@ -316,6 +317,35 @@ def mutate(
...
+
+    @capture_error
+    @require_permission_check(Permissions.LAUNCH_PIPELINE_EXECUTION)
+    def mutate(
+        self, graphene_info: ResolveInfo, executionParamsList: List[GrapheneExecutionParams]
+    ):
+        launch_multiple_runs_result = []
+
+        for execution_params in executionParamsList:
+            result = GrapheneLaunchRunMutation.mutate(
```
Comment (salazarm) [inline python_modules/dagster-graphql/dagster_graphql/schema/roots/mutation.py:339]: I didn't even know you could do this tbh. Is this kosher @alangenfeld ?

Comment (salazarm) [reply python_modules/dagster-graphql/dagster_graphql/schema/roots/mutation.py:339]: ah yeah I see this pattern used in a couple other place

File: python_modules/dagster-graphql/dagster_graphql/schema/runs.py [hunk 3]
```diff
@@ -73,17 +74,22 @@ class Meta:
 
 class GrapheneLaunchRunResult(graphene.Union):
     class Meta:
-        from dagster_graphql.schema.backfill import pipeline_execution_error_types
```
Comment (salazarm) [inline python_modules/dagster-graphql/dagster_graphql/schema/runs.py:76]: why was this removed?

Comment (salazarm) [reply python_modules/dagster-graphql/dagster_graphql/schema/runs.py:76]: and why was it there in the first place 🤔

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_run_launcher.py [hunk 4]
```diff
@@ -83,10 +93,127 @@ def test_run_launcher_subset(self, graphql_context: WorkspaceRequestContext):
...
+            run_ids.append(result["run"]["runId"])
+
+        wait_for_runs_to_finish(graphql_context.instance)
+
+        for run_id in run_ids:
+            result = execute_dagster_graphql(
+                context=graphql_context, query=RUN_QUERY, variables={"runId": run_id}
+            )
+            assert result.data["pipelineRunOrError"]["__typename"] == "Run"
+            assert result.data["pipelineRunOrError"]["status"] == "SUCCESS"
```
Comment (salazarm) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_run_launcher.py:146]: not sure we need this? I guess it's just asserting that the runs would have succeeded also if they were run individually but I think we can just assume that is the case given the mock setup.

File: python_modules/dagster-graphql/dagster_graphql/schema/roots/mutation.py [hunk 5]
```diff
@@ -316,6 +317,35 @@ def mutate(
...
+    Output = graphene.NonNull(GrapheneLaunchMultipleRunsResult)
+
+    class Arguments:
+        executionParamsList = non_null_list(GrapheneExecutionParams)
+
+    class Meta:
+        name = "LaunchMultipleRunsMutation"
+
+    @capture_error
+    @require_permission_check(Permissions.LAUNCH_PIPELINE_EXECUTION)
```
Comment (salazarm) [inline python_modules/dagster-graphql/dagster_graphql/schema/roots/mutation.py:332]: This permission check here can make the whole mutation fail, and in this case the output would be a single GrapheneUnauthorizedError so I believe your output should be a union of `GrapheneLaunchMultipleRunsResult` and `GrapheneUnauthorizedError` and also `GraphenePythonError` since we're using `@capture_error`.

**Review by salazarm — CHANGES_REQUESTED** [review]
Fix the output type to be a union of GraphenePythonError, GrapheneUnauthorizedError and GrapheneLaunchMultipleRunsResult

File: python_modules/dagster-graphql/dagster_graphql/schema/runs.py (hunk 3 above)
Comment (dliu27) [reply python_modules/dagster-graphql/dagster_graphql/schema/runs.py:76]: Not sure but I moved the import to top of file: https://github.com/dagster-io/dagster/pull/25880/files#diff-d6694e6d32ed9271ea5c36951e11ae6f342d707257fb8056db67e50df6d69d14R14
